contracts/SpeedMarkets/SpeedMarketsAMMUtils.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/// @title An AMM utils for Thales speed markets
+contract SpeedMarketsAMMUtils {
+    uint private constant SECONDS_PER_MINUTE = 60;
+
+    /// @notice get dynamic fee based on defined time thresholds for a given delta time
+    /// @param _deltaTimeSec to search for appropriate time range (in seconds)
+    /// @param _timeThresholds array of time thresholds for each fee (in minutes)
+    /// @param _fees array of fees for every time range
+    /// @param _defaultFee if _deltaTime doesn't have appropriate time range return this value
+    /// @return fee defined for specific time range to which _deltaTime belongs to
+    function getFeeByTimeThreshold(
+        uint64 _deltaTimeSec,
+        uint[] calldata _timeThresholds,
+        uint[] calldata _fees,
+        uint _defaultFee
+    ) public pure returns (uint fee) {
+        require(_timeThresholds.length == _fees.length, "Time thresholds and fees arrays must have the same length");
```

Comment:
shouldnt this be guaranteed with the setter?

Reply:
Yes, intentionally put it to protect from inside, but I'll remove it.